controllers/helpers.go

```diff
@@ -955,6 +955,10 @@ func MachinePoolToAzureManagedControlPlaneMapFunc(ctx context.Context, c client.
 
 		gk := gvk.GroupKind()
 		ref := cluster.Spec.ControlPlaneRef
+		if ref == nil || ref.Name == "" {
+			log.Info("control plane ref is nil or empty: failed to get the control plane ref")
```

Comment:
an info log should contained the word "failed", so probably this either needs to be an error log or it should be reworded if it's purely informative